models/events.go

```diff
@@ -70,13 +70,25 @@ func (c *CadenceEvents) Blocks() ([]*types.Block, error) {
 func (c *CadenceEvents) Transactions() ([]Transaction, []*StorageReceipt, error) {
 	txs := make([]Transaction, 0)
 	rcps := make([]*StorageReceipt, 0)
+
+	cumulativeGasUsed := uint64(0)
+	var lastReceipt *StorageReceipt
 	for _, e := range c.events.Events {
 		if isTransactionExecutedEvent(e.Value) {
 			tx, receipt, err := decodeTransactionEvent(e.Value)
 			if err != nil {
 				return nil, nil, err
 			}
 
+			if lastReceipt != nil && lastReceipt.BlockHash == receipt.BlockHash {
+				cumulativeGasUsed += lastReceipt.GasUsed
+				receipt.CumulativeGasUsed = cumulativeGasUsed
```

Comment:
the first receipt doesn't have the cumulative gas set to the gas used value

Reply:
or actually it will have just by chance because we match cumulative gas to gas used in decode receipt, I don't like this code relies on that code, it would be better to improve this code

Reply:
> or actually it will have just by chance because we match cumulative gas to gas used in decode receipt, I don't like this code relies on that code, it would be better to improve this code

That's a good catch :100: 
Updated in https://github.com/onflow/flow-evm-gateway/pull/358/commits/d6628b3f0354c6ec051a2a6aa0fb188f54459f59
I have removed the default assignment of `CumulativeGasUsed` from `decodeTransactionEvent`, and added some more test assertions to make sure that we handle cases with blocks that have multiple transactions, and blocks with single transactions.